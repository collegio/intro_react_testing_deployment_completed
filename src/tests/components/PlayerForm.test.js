import React from 'react';
import { shallow } from 'enzyme';
import PlayerForm from '../../components/PlayerForm';
import players from '../fixtures/players';

test('should render the player form', () => {
    const wrapper = shallow(<PlayerForm />);
    expect(wrapper).toMatchSnapshot();
});

test('should render the player form with data', () => {
    const wrapper = shallow(<PlayerForm player={players[0]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render error for invalid form submission', () => {
    const wrapper = shallow(<PlayerForm />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(wrapper.state('error').length).toBeGreaterThan(0);
    expect(wrapper).toMatchSnapshot();
});

test('should set name on input change', () => {
    const value = 'Sidney Crosby';
    const wrapper = shallow(<PlayerForm />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('input').simulate('change', {
        target: { value }
    });
    expect(wrapper.state('name')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should set sport type on select change', () => {
    const value = 'baseball';
    const wrapper = shallow(<PlayerForm />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('select').at(0).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('sport_type')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should set gender on select change', () => {
    const value = 'Female';
    const wrapper = shallow(<PlayerForm />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('select').at(1).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('gender')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should set skill level on select change', () => {
    const value = 'expert';
    const wrapper = shallow(<PlayerForm />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('select').at(2).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('skill_level')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should trigger valid form submission', () => {
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(<PlayerForm player={players[0]} onSubmit={onSubmitSpy}/>);

    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });

    expect(wrapper.state('error')).toBe('');
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        name: players[0].name,
        sport_type: players[0].sport_type,
        gender: players[0].gender,
        skill_level: players[0].skill_level,
        message: players[0].message
    });
});