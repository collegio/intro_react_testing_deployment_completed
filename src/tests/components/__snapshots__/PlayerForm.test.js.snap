// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render error for invalid form submission 1`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should render error for invalid form submission 2`] = `
<div
  className="player-form"
>
  <p>
    Please enter a name first.
  </p>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should render the player form 1`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should render the player form with data 1`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Sidney Crosby"
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="expert"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value="Just looking to play"
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should set gender on select change 1`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should set gender on select change 2`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Female"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should set name on input change 1`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should set name on input change 2`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Sidney Crosby"
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should set skill level on select change 1`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should set skill level on select change 2`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="expert"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should set sport type on select change 1`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="hockey"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should set sport type on select change 2`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <br />
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <label>
      Sport
    </label>
    <br />
    <select
      onChange={[Function]}
      value="baseball"
    >
      <option
        value="hockey"
      >
        Hockey
      </option>
      <option
        value="baseball"
      >
        Baseball
      </option>
      <option
        value="softball"
      >
        Softball
      </option>
      <option
        value="basketball"
      >
        Basketball
      </option>
    </select>
    <label>
      Gender
    </label>
    <br />
    <select
      onChange={[Function]}
      value="Male"
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
    </select>
    <label>
      Skill Level
    </label>
    <br />
    <select
      onChange={[Function]}
      value="basic"
    >
      <option
        value="basic"
      >
        Basic
      </option>
      <option
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        value="expert"
      >
        Expert
      </option>
    </select>
    <label>
      Message
    </label>
    <br />
    <textarea
      onChange={[Function]}
      placeholder="Message"
      value=""
    />
    <button
      className="button pull-right"
    >
      Submit
    </button>
  </form>
</div>
`;
